### ISSUE TITLE
[Android] App Crash Right on Launch : NullPointerException 

### ISSUE THREAD

**Issue by dio-ralali** [launchdarkly/react-native-client-sdk#205]
i've used launchdarkly since Sept 2022, and haven't changed it version ever since. Always run smoothly until this day. When im build the release apk for testing, the app is crashed right on launch. 

### **Issue only occures on Android on Release APK. On iOS and Android when debugging its all works fine as intended.**

`com.launchdarkly.reactnative.LaunchdarklyReactNativeClientModule$ConfigMapping.loadFromMap (LaunchdarklyReactNativeClientModule.java:794)
com.launchdarkly.reactnative.LaunchdarklyReactNativeClientModule.configBuild (LaunchdarklyReactNativeClientModule.java:288)
com.launchdarkly.reactnative.LaunchdarklyReactNativeClientModule.internalConfigure (LaunchdarklyReactNativeClientModule.java:255)
com.launchdarkly.reactnative.LaunchdarklyReactNativeClientModule.configure (LaunchdarklyReactNativeClientModule.java:230)
java.lang.reflect.Method.invoke (Method.java)
com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:151)
com.facebook.react.bridge.queue.NativeRunnable.run (NativeRunnable.java)
android.os.Handler.handleCallback (Handler.java:938)

android.os.Handler.dispatchMessage (Handler.java:99)
com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
android.os.Looper.loopOnce (Looper.java:201)

android.os.Looper.loop (Looper.java:288)
com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
`

![CleanShot 2023-03-14 at 15 09 04](https://user-images.githubusercontent.com/86591816/224936184-036d2fa0-6fc2-4384-84e4-9ff15ff274cb.png)

**my package.json : 
    "react-native": "^0.64.4",
    "launchdarkly-react-native-client-sdk": "^6.2.0",**

**Comment by denissb** [comment]
I second that, tried to upgrade the SDK and faced the same issue on Android.
Please not it happens only in release mode, in debug it starts just fine.

**Comment by PunainenAurinko** [comment]
Same is happening to me with "launchdarkly-react-native-client-sdk": "^7.1.1", it works on iOS but crashes on app launch on Android in release mode. Works in debug on both platforms.

**Comment by vlad-ps** [comment]
Same is happening to me with "launchdarkly-react-native-client-sdk": "^7.1.1", it works on iOS but crashes on app launch on Android in release mode. Works in debug on both platforms.

**Comment by RobertWHurst** [comment]
My team is also facing the same exception. No idea why as I don't think we changed anything in regards to the configuration or version of the package.

**Comment by dio-ralali** [comment]
does anyone have temporary solutions beside disabling launchdarkly? my team really need to releasing new feature in a few days.

**Comment by louis-launchdarkly** [comment]
Hello @dio-ralali, @denissb, @PunainenAurinko, @vlad-ps, @RobertWHurst, 

Thank you for reporting the bug, we need more information to investigate this.

If it is possible, can you please share your RN SDK release build and debug build configuration (while removing all the secret values) so we can reproduce this?

**Comment by RobertWHurst** [comment]
@louis-launchdarkly As far as versions go, we are using React Native version 0.68.5 and Launch Darkly 6.3.2.

**Comment by louis-launchdarkly** [comment]
@RobertWHurst Thank you for the info, we suspect something had changed at 6.3.x. If you can share the hash/object you used as a configuration to initialize the LDClient, that would be really helpful for reproduction.

**Comment by vlad-ps** [comment]
We were using LD before, but it started crashing recently. Currently we are on v. 7.1.1 with RN 0.71.4. 

That's what we used to initialize LD:

```
import LDClient, * as ld from 'launchdarkly-react-native-client-sdk';

  // * Initialize LaunchDarkly
     useEffect(() => {
         const client = new LDClient()

         if (isLoggedIn) {
             const { apiUrlKey } = session // session object unrelated to LD

             const mobileKey = CONFIG[apiUrlKey]?.LD_KEY || CONFIG.PROD.LD_KEY // mobile key for different environments that we use for the app

             const config: ld.LDConfig = { mobileKey }
             const context: ld.LDContext = {
                 key: `user-key-${CLIENT_ID}`,
                 kind: 'user',
             }

             const initialize = async () =>
                 await client.configure(config, context)

             initialize().catch((error) => console.error(error))
         }
         return async () => await client.close()
     }, [])
```

**Comment by RobertWHurst** [comment]
@louis-launchdarkly Here's a snippet of our code for initializing the Launch Darkly client:
```

export const configure = async () => {
  const state = store.getState();
  const config = {
    mobileKey:
      getEnv('ENV') === 'prod'
        ? getProdKey()
        : getStageKey(),
  };
  const user = {
    key: state.account.currentUser?.auth0Id?.replace('|', '-'),
    email: state.account.currentUser?.email,
    firstName: state.account.currentUser?.givenName,
    lastName: state.account.currentUser?.familyName,
  };

  try {
    await client.close();
    await client.configure(config, user);

    // Set up feature flags
    const dataCreditRedemptionEnabled = await client.stringVariation('data-credit-redemption');
    store.dispatch(updateDataCreditRedemptionEnabled(dataCreditRedemptionEnabled));

    const pushNotificatonsEnabled = await client.stringVariation('push-notifications');
    store.dispatch(updatePushNotificationsEnabled(pushNotificatonsEnabled));

    console.log({
      dataCreditRedemptionEnabled,
      pushNotificatonsEnabled,
    });
  } catch (e) {
    // no-op
    console.log(e);
  }
};


```

**Comment by yusinto** [comment]
@RobertWHurst the user key property is mandatory for the React Native SDK versions 6.x because the underlying Android SDK mandates it. In versions >= 7.x the React Native SDK will auto generate the context key if none is provided. Please update to the latest 7.x version to resolve this issue.

If you have to stay on versions 6.x, the [workaround](https://github.com/launchdarkly/react-native-client-sdk/issues/179#issuecomment-1441192633) is to ensure that the user key is set.

**Comment by RobertWHurst** [comment]
@yusinto Thank for that. Much appreciated, we'll give it a shot 🍻

**Comment by RobertWHurst** [comment]
@yusinto Indeed we do use optional chaining there, but I can confirm we always set the key. It is never an empty value. That said, just to be sure we will update the code to only allow configure to be called if currentUser is present.

**Comment by yusinto** [comment]
@dio-ralali the [error line](https://github.com/launchdarkly/react-native-client-sdk/blob/5f3be705ce8b30cd7da1c82f8e342f96c0e1a874/android/src/main/java/com/launchdarkly/reactnative/LaunchdarklyReactNativeClientModule.java#L794) from your screenshot suggests that you are using version 6.3.x. If you set the rn sdk dependency to`^6.2.0` then the ld rn sdk will get updated to 6.3.x.

As a workaround for now, you can pin your dependency to 6.2.x while we investigate the issue:

```json
    "launchdarkly-react-native-client-sdk": "~6.2.0",
```

**Comment by louis-launchdarkly** [comment]
Hello all,

We are still looking into this problem. In the meantime, @dio-ralali, please let us know does pinning the dependency help resolve the problem for you for now while we are investigating this.

**Comment by RobertWHurst** [comment]
@louis-launchdarkly @yusinto I can confirm that 6.2.x do not cause the issue. We rolled back and that did sort it out for us for now.

**Comment by dio-ralali** [comment]
Yep i can confirm that pinning to version "~6.2.0" did work for me. Thanks for the solution 👍

**Comment by denissb** [comment]
@louis-launchdarkly

This is how we initialise the LD SDK:
```
export async function initLdClient(mobileKey: string) {
	const client = new LDClient();

	let config = {
		mobileKey, // you know what this is
		stream: true,
		offline: false,
		connectionTimeoutMillis: TIMEOUT_MS, // 5000
	};

	let user = {
		key: await DeviceInfo.getUniqueId(), // use device unique ID as anonymous user ID
		anonymous: true,
	};

	try {
		await client.configure(config, user, TIMEOUT_MS / 1000);
		log.log('Initialized LD client');
	} catch (err) {
		if (
			err instanceof Error &&
			err.message === 'LaunchDarkly SDK already initialized'
		) {
			// Do nothing
		} else if (
			err instanceof Error &&
			err.message === 'SDK initialization timed out'
		) {
			log.warn(err);
			Analytics.trackEvent('LaunchDarkly configuration timed out', {
				error: String(err),
			});

			log.log(
				'Trying to configure LaunchDarkly silently without blocking',
			);

			client
				.configure(config, user)
				.then(() => {
					log.log('LaunchDarkly configured successfully after retry');
				})
				.catch(() => null);
		} else {
			log.warn(err);
			Analytics.trackEvent('Failed to initialized LaunchDarkly', {
				error: String(err),
			});
		}
	}

	return client;
}
```

**"react-native": "0.66.5",
"launchdarkly-react-native-client-sdk": "6.3.1"**

**Comment by yusinto** [comment]
We have released [7.1.2](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/7.1.2) and [6.3.3](https://github.com/launchdarkly/react-native-client-sdk/releases/tag/6.3.3) which fix the loadFromMap configBuild errors as reported by @dio-ralali.

**Comment by dio-ralali** [comment]
Thanks a lot mas @yusinto! I can confirm that the 6.3.3 version already fixed the issue. And i can remove the pinning dependency and back to this package.json config : 

"react-native": "^0.64.4",
"launchdarkly-react-native-client-sdk": "^6.2.0",
